main.tf

```diff
@@ -101,7 +108,7 @@ resource "azuredevops_build_definition" "build_definition" {
     branch_name           = var.repository.branch_name
     repo_id               = var.repository.repo_type == "TfsGit" ? data.azuredevops_git_repository.repo[0].id : var.repository.repo_id
     repo_type             = var.repository.repo_type
-    service_connection_id = var.repository.service_connection_id
```

Comment:
looks like now we have `var.repository.service_connection_id` not being used, and `var.service_connection_name` is used instead. 

Im guessing the original intention with `var.repository.service_connection_id` was to specify the name, since the actual IDs of these are pretty hidden. I suggest we get rid of `var.service_connection_name` and use the existing `var.repository.service_connection_id` to specify the name instead. this way we can avoid changing the reference module. 

at least thats what I assumed when I did this with `var.repository.repo_id` a few weeks ago

Reply:
I have removed service_connection_name and modified the code using service_connection_id